src/main/java/edu/illinois/cs/dt/tools/detection/detectors/ReverseDetector.java

```diff
@@ -23,8 +24,13 @@ public ReverseDetector(final Runner runner, final int rounds, final String name,
 
         testShuffler = new TestShuffler(name, rounds, tests);
 
+        // Filters to be applied in order
+        if (runner instanceof InstrumentingSmartRunner) {
+            addFilter(new ConfirmationFilter(name, tests, (InstrumentingSmartRunner) runner));
+        } else {
+            addFilter(new ConfirmationFilter(name, tests, InstrumentingSmartRunner.fromRunner(runner)));
+        }
         addFilter(new UniqueFilter());
-        addFilter(new VerifyFilter(name, runner));
```

Comment:
I see that the `VerifyFilter` eventually calls the `verify` method in `ConfirmationFilter`. Does the order in which the `verify` method gets called matter? (Prior to these changes, the `verify` method is called after `UniqueFilter`. With the changes, the `verify` method will be called before.)

Reply:
Ultimately, I don't think the order matters, the filters should be commutative. In RandomDetector, the ConfirmationFilter comes first, so I was copying that style here.